src/BuildScriptGenerator/DefaultBuildScriptGenerator.cs

```diff
@@ -162,13 +183,23 @@ internal class DefaultBuildScriptGenerator : IBuildScriptGenerator
 
         private IList<BuildScriptSnippet> GetBuildSnippets(
             BuildScriptGeneratorContext context,
+            IEnumerable<PlatformDetectorResult> detectionResults,
+            bool runDetection,
```

Comment:
Here does `runDetection` mean detect from all platforms? If so, seems the logic here is opposite..